core/trino-main/src/main/java/io/trino/operator/MultiChannelGroupByHash.java

```diff
@@ -984,51 +986,56 @@ public GroupByIdBlock getResult()
             implements Work<T>
     {
         protected final Page page;
-        protected final int maxCardinality;
-        protected final int[] dictionarySizes;
-        protected final DictionaryBlock[] blocks;
 
         public LowCardinalityDictionaryWork(Page page)
         {
             this.page = requireNonNull(page, "page is null");
-            dictionarySizes = new int[channels.length];
-            blocks = new DictionaryBlock[channels.length];
-            int maxCardinality = 1;
-            for (int i = 0; i < channels.length; i++) {
-                Block block = page.getBlock(channels[i]);
-                verify(block instanceof DictionaryBlock, "Only dictionary blocks are supported");
-                blocks[i] = (DictionaryBlock) block;
-                int blockPositionCount = blocks[i].getDictionary().getPositionCount();
-                dictionarySizes[i] = blockPositionCount;
-                maxCardinality *= blockPositionCount;
-            }
-            this.maxCardinality = maxCardinality;
         }
 
         /**
          * Returns combinations of all dictionaries ids for every position and populates
          * samplePositions array with a single occurrence of every used combination
          */
-        protected short[] calculateCombinationIdsToPositionMapping(int[] combinationIdToPosition)
+        protected int[] calculateCombinationIdToPosition(Page page)
```

Comment:
actually, the best name would be `calculateCombinationIdToPositionMapping`

Reply:
Renamed.